### ISSUE TITLE
Resume playback not working

### ISSUE THREAD

**Issue by mejico** [anxdpanic/plugin.video.youtube#904]
### Context

Please provide any relevant information about your setup

* Add-on Version: v7.0.9.2
* Kodi Version: 19.4
* Kodi GUI Language: English
* Operating System: Windows 11
* Operating System Language: English

------

### Expected Behaviour: 

In the past, when you stopped playing a video and went back to it sometime later it would give you the option to resume play from the time stamp of when you stopped the video.

------

### Current Behaviour

When you stopped playing a video and return to it another time you only have the option to playback the video from the start.

------

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Play a video until halfway (assuming the video is longer than 3 minutes long).
2. Stop the video.
3. Play the same video again.

------

### Log

[kodi Log.txt](https://github.com/user-attachments/files/17053398/kodi.Log.txt)

------

### Additional Information

This feature was working fine, up until about a week prior to the latest release.

------

**Comment by MoojMidge** [comment]
Can you get a debug log of playing, stopping then trying to resume a video? 

https://kodi.wiki/view/Log_file/Easy

A standard log does not have enough information to determine what the problem may be.

**Comment by mejico** [comment]
Will do as soon as I get a chance.  Thanks for the reply.  

I also should have mentioned that prior to raising the bug report I asked about this problem on the Kodi YouTube plugin thread but the only reply I got was that someone else had the same problem, but unfortunately no solution.

**Comment by mejico** [comment]
Sorry about the delay.  Here is the debug log.

Cheers
[kodi Debug log.txt](https://github.com/user-attachments/files/17073786/kodi.Debug.log.txt)

**Comment by mejico** [comment]
I have noticed that if I restart Kodi, the resume function will work for the first video I play, but will not work for every subsequent video until Kodi is restarted.

**Comment by MoojMidge** [comment]
Can you test if the problem is fixed when using this test version?
https://github.com/MoojMidge/plugin.video.youtube/archive/refs/heads/master.zip

**Comment by mejico** [comment]
Will give it a go and get back to you

Cheers

**Comment by mejico** [comment]
That seems to have fixed the problem.  
Was the problem caused by changes made by YouTube,?

**Comment by MoojMidge** [comment]
Older versions of Kodi have issues with resuming from plugins. This used to be worked around to make it work in this plugin, but has now been largely fixed in more recent versions of Kodi and the old workarounds conflicted with the native Kodi functionality.

This was fixed to work with the native functionality, but this caused a regression on the older versions that still rely on this old behaviour.

TLDR: Upgrade Kodi, there are lots of improvements and it is a lot of work to keep testing how things work on these older releases.
